.nextmv/update_apps.py

```diff
@@ -126,20 +141,59 @@ def get_manifest(
 ) -> dict[str, Any]:
     """Returns the manifest from the S3 bucket."""
 
-    result = client.get_object(Bucket=bucket, Key=f"{folder}/{manifest_file}")
-    return yaml.safe_load(result["Body"].read())
+    log("Getting manifest from S3 bucket.")
 
+    manifest = {}
+    try:
+        result = client.get_object(Bucket=bucket, Key=f"{folder}/{manifest_file}")
+        manifest = yaml.safe_load(result["Body"].read())
+        log("Obtained manifest from S3 bucket.")
 
-def tar_app(name: str, version: str) -> str:
-    """Create a tarbal of the app. Returns the name of the tarball."""
+    except client.exceptions.NoSuchKey:
+        log("No manifest in S3 bucket.")
+        pass
 
-    app_dir = os.path.join(os.getcwd(), "..", name)
-    filename = f"{name}_{version}.tar.gz"
+    return manifest
 
-    with tarfile.open(filename, "w:gz") as tar:
-        tar.add(app_dir, arcname=os.path.basename(app_dir))
 
-    return filename
+def log(message: str):
+    """Logs a message to the console."""
+
+    print(f"🐰 {message}", file=sys.stdout, flush=True)
+
+
+def push_app(name: str, version: str):
+    """Pushes the app to the Nextmv Marketplace."""
+
+    log(f"Pushing app: {name}; version: {version} to the Marketplace.")
+
+    workflow_info = app_workflow_info(name)
+    app_id = workflow_info.get("app_id") or None
+    marketplace_app_id = workflow_info.get("marketplace_app_id", "") or None
+    if app_id is None or marketplace_app_id is None:
+        log(f"App: {name} has not app_id or marketplace_app_id, skipping push to Cloud.")
+        return
```

Comment:
Nitpick: wording on log is not immediately clear to me. "has not app_id or..." makes the "has not" part look like it only applies app_id at first glance. If I were to reword it I'd do so as follows (I also change a comma to a semicolon here as these are independent clauses closely related by logical follow-through):
```suggestion
    if app_id is None or marketplace_app_id is None:
        log(f"App: {name} either app_id or marketplace_app_id is not defined; skipping push to Cloud.")
        return
```